pkg/da/data_availability_header.go

```diff
@@ -62,9 +65,39 @@ func NewDataAvailabilityHeader(eds *rsmt2d.ExtendedDataSquare) (DataAvailability
 	return dah, nil
 }
 
+// ConstructEDS constructs an ExtendedDataSquare from the given transactions and app version.
+// If maxSquareSize is less than 0, it will use the upper bound square size for the given app version.
+func ConstructEDS(txs [][]byte, appVersion uint64, maxSquareSize int) (*rsmt2d.ExtendedDataSquare, error) {
+	if appVersion == 0 {
+		return nil, fmt.Errorf("app version cannot be 0")
+	}
+	if appVersion <= v5.Version {
+		if maxSquareSize < 0 {
+			maxSquareSize = v5.SquareSizeUpperBound
+		}
+		// all versions 5 and below have the same parameters and algorithm
+		square, err := squarev2.Construct(txs, maxSquareSize, v5.SubtreeRootThreshold)
```

Comment:
Is this true? I thought app version 3 which introduced authored blobs had a different construct method which led to https://github.com/celestiaorg/celestia-app/blob/4dd92890ac0a50d163627053d6e0323c9b5ba6a9/app/process_proposal.go#L140-L158

on re-review there is also https://github.com/celestiaorg/celestia-app/issues/4690 so maybe we don't need cases for the historical app versions and go-versions.

Reply:
Wait @cmwaters wasn't the whole problem with the broken v2 that it did not encode authored blobs correctly and that's why we need v3?